RecoHGCal/TICL/plugins/TICLCandidateFromTrackstersProducer.cc

```diff
@@ -0,0 +1,143 @@
+// user include files
+#include <algorithm>
+#include <vector>
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+#include "FWCore/Utilities/interface/transform.h"
+
+#include "DataFormats/HGCalReco/interface/TICLCandidate.h"
+#include "DataFormats/HGCalReco/interface/Trackster.h"
+#include "RecoHGCal/TICL/interface/TracksterMomentumPluginBase.h"
+#include "RecoHGCal/TICL/interface/TracksterTrackPluginBase.h"
+
+using namespace ticl;
+
+class TICLCandidateFromTrackstersProducer : public edm::stream::EDProducer<> {
+public:
+  TICLCandidateFromTrackstersProducer(const edm::ParameterSet&);
+  ~TICLCandidateFromTrackstersProducer() override {}
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  void produce(edm::Event&, const edm::EventSetup&) override;
+
+private:
+  std::vector<edm::EDGetTokenT<std::vector<Trackster>>> trackster_tokens_;
+  std::unique_ptr<TracksterMomentumPluginBase> momentum_algo_;
+  std::unique_ptr<TracksterTrackPluginBase> track_algo_;
+  float min_particle_prob_;
+};
+DEFINE_FWK_MODULE(TICLCandidateFromTrackstersProducer);
+
+namespace {
+  int pdg_id_from_idx(size_t i) {
+    switch (i) {
+      case 0:
```

Comment:
use the actual enums for these case values if possible

Reply:
Done and fully agreed that this is better.